src/main/java/com/mysema/maven/apt/AbstractProcessorMojo.java

```diff
@@ -279,6 +282,42 @@ private String buildProcessor() {
         return files;
     }
 
+    /**
+     * Add messages through the buildContext:
+     * <ul>
+     *   <li>cli build creates log output</li>
+     *   <li>m2e build creates markers for eclipse</li>
+     * </ul>
+     * @param diagnostics
+     */
+    private void processDiagnostics(final List<Diagnostic<? extends JavaFileObject>> diagnostics) {
+        for (Diagnostic<? extends JavaFileObject> diagnostic : diagnostics) {
+            if (diagnostic != null) {

```

Comment:
`Diagnostic`s are not supposed to be null, so you can rely on that fact.
This null check is not necessary.